equational_theories/EquationalResult.lean

```diff
@@ -87,35 +92,61 @@ def extractEquationalResults {m : Type → Type} [Monad m] [MonadEnv m] [MonadEr
     m (Array Entry) := do
   return equationalResultsExtension.getState (← getEnv)
 
+def extractTheorems {m : Type → Type} [Monad m] [MonadEnv m] [MonadError m] :
+    m (Array Entry) := do
+  return (equationalResultsExtension.getState (← getEnv)).filter (·.proven)
+
+def extractConjectures {m : Type → Type} [Monad m] [MonadEnv m] [MonadError m] :
+    m (Array Entry) := do
+  return (equationalResultsExtension.getState (← getEnv)).filter (!·.proven)
+
 /-- Prints the contents of the equational results environment extension.
 -/
 syntax (name := printEquationalResults) "#print_equational_results" : command
 
 elab_rules : command
 | `(command| #print_equational_results) => do
-  let rs ← extractEquationalResults
-  for ⟨name, _filename, res⟩ in rs do
+  let rs ← extractTheorems
+  for ⟨name, _filename, res, _⟩ in rs do
     match res with
     | .implication ⟨lhs, rhs⟩ => println! "{name}: {lhs} → {rhs}"
     | .facts ⟨satisfied, refuted⟩ => println! "{name}: {satisfied} // {refuted}"
     | .unconditional rhs => println! "{name}: {rhs} holds unconditionally"
 
---- Output of the extract_implications executable.
-structure Output where
-  implications : Array Implication
-  facts : Array Facts
-  unconditionals : Array String
-deriving Lean.ToJson, Lean.FromJson
-
-def collectResults {m : Type → Type} [Monad m] [MonadEnv m] [MonadError m] :
-    m Output := do
-  let rs := equationalResultsExtension.getState (← getEnv)
-  let mut implications : Array Implication := #[]
-  let mut facts : Array Facts := #[]
-  let mut unconditionals : Array String := #[]
-  for ⟨_name, _filename, res⟩ in rs do
-    match res with
-    | .implication imp => implications := implications.push imp
-    | .facts fact => facts := facts.push fact
-    | .unconditional s => unconditionals := unconditionals.push s
-  return ⟨implications, facts, unconditionals⟩
+end Result
+
+namespace Conjecture
+
+/--
+Like `proof_wanted` from Batteries, but records the conjecture in an
+environment extension.
```

Comment:
This comment should be updated. It only records the conjecture if it has been tagged with `@[equational_result]`.